Update to Postsrsd 2.x

FingerlessGlov3s opened the pull request:
I use this container for my Mailcow Email Server to carry out the SRS function, recently this container got updated when I did a `docker compose pull`, which then broke my email delivery. I'm assuming you have a automatic pipeline to rebuild your containers every month or so, to keep them up to date and have not noticed this is currently broken.

The upstream package from alpine changed from Postsrsd 1.x to Postsrsd 2.x, which has broken this container. I have updated the `Dockerfile` and `run` file to bring it up to line with the postsrsd 2.x changes.

You will also need to update your postfix config from using tcp to socketmap for the canonical_maps. I changed the postsrsd port from 1001-1002 to 1003, to make it obvious it's a different protocol.

Below are the changes I also made to my postfix environment to get it working with the new version of postsrsd. Hopefully you will accept my merge request and we can get this container working again nicely.

####  data/conf/postfix/extra.cf
```
recipient_canonical_maps = tcp:172.22.1.42:10002, proxy:mysql:/opt/postfix/conf/sql/mysql_recipient_canonical_maps.cf
## changed to 
recipient_canonical_maps = socketmap:inet:172.22.1.42:10003:reverse, proxy:mysql:/opt/postfix/conf/sql/mysql_recipient_canonical_maps.cf
```


####  data/conf/postfix/master.cf
```
-o sender_canonical_maps=tcp:172.22.1.42:10001
## changed to
-o sender_canonical_maps=socketmap:inet:172.22.1.42:10003:forward
```

FingerlessGlov3s committed:
Update to Postsrsd 2.x

FingerlessGlov3s committed:
syslog was making it log twice, this fixes that issue

FingerlessGlov3s commented:
I changed the syslog to off and make it so it prepends a datetime for every outputted line for postsrsd.

Otherwise you get double messages, I was going to redirect stderr to `/dev/null` but then if there's a configuration error it doesn't log, so makes it harder to debug why it's not working.

Example
```
2023-05-20 00:11:41 - Starting postsrsd
2023-05-20 00:12:08 - postsrsd: <mytestemail@gmail.com> not reversed: Not an SRS address.
2023-05-20 00:12:08 - postsrsd: <test@domain.uk> not reversed: Not an SRS address.
2023-05-20 00:12:08 - postsrsd: <test@domain.uk> not reversed: Not an SRS address
```

RedJohn14 commented:
Hello everyone,

sorry for reopen this Issue. But after upgrade today I have the same problem again :-(. I have set the configuration like this: 

**/etc/postfix/main.cf**

`sender_canonical_maps = socketmap:unix:srs:forward
sender_canonical_classes = envelope_sender
recipient_canonical_maps = socketmap:unix:srs:reverse
recipient_canonical_classes = envelope_recipient, header_recipien`

Is there no change needed in **master.conf?** Before I have this settings in my master.conf:

```
cleanup-srs unix  n       -       -       -       0       cleanup
      -o sender_canonical_maps=tcp:172.22.1.42:10001
      -o sender_canonical_classes=envelope_sender
      -o recipient_canonical_maps=regexp:/opt/postfix/conf/regex_sender_canonical_srs

127.0.0.1:10029 inet    n       -       -       -       -       smtpd
        -o cleanup_service_name=cleanup-srs
        -o smtpd_tls_security_level=none
        -o content_filter=smtp:
        -o smtpd_recipient_restrictions=permit_mynetworks,reject
        -o smtpd_milters=
```

If I change the line **-o sender_canonical_maps=tcp:172.22.1.42:10001** to **-o sender_canonical_maps=socketmap:inet:172.22.1.42:10003:forward** and use the settings above in the extra.conf I have problems in the log like this:

```
Jun 19 22:46:38 postfix/cleanup[391]: warning: table socketmap:unix:srs:forward lookup error: No such file or directory
Jun 19 22:46:38 postfix/cleanup[391]: warning: socketmap:unix:srs:forward lookup error for "double-bounce@mydomain.de"
Jun 19 22:46:38 postfix/cleanup[391]: warning: B272580380: sender_canonical_maps map lookup problem for double-bounce@mydomain.de-- message not accepted, try again later
```

Sorry I hope you can help me!

Thanks!

FingerlessGlov3s commented:
Your trying to use unix socket, you need to use inet for all parts of the postsrsd.

Double check what you have, comparing it to working config posted above.

Make sure you do both `extra.cf` and `master.cf`

RedJohn14 commented:
> Your trying to use unix socket, you need to use inet for all parts of the postsrsd.
> 
> Double check what you have, comparing it to working config posted above.
> 
> Make sure you do both `main.cf` and `master.cf`

Yes I have double checked my config everything is correct. Can you send the part from your config and I can compare it. At the moment I rolled back to the old docker image for working mailflow. :-(

FingerlessGlov3s commented:
I have added the following below to both my `extra.cf` and `master.cf`

Added to the bottom of `data/conf/postfix/extra.cf`
```
sender_canonical_classes = envelope_sender
recipient_canonical_maps = socketmap:inet:172.22.1.42:10003:reverse, proxy:mysql:/opt/postfix/conf/sql/mysql_recipient_canonical_maps.cf
recipient_canonical_classes = envelope_recipient, header_recipient
```

Added to the bottom of `data/conf/postfix/master.cf`
```
cleanup-srs unix  n       -       -       -       0       cleanup
      -o sender_canonical_maps=socketmap:inet:172.22.1.42:10003:forward
      -o sender_canonical_classes=envelope_sender
      -o recipient_canonical_maps=regexp:/opt/postfix/conf/regex_sender_canonical_srs

127.0.0.1:10029 inet    n       -       -       -       -       smtpd
        -o cleanup_service_name=cleanup-srs
        -o smtpd_tls_security_level=none
        -o content_filter=smtp:
        -o smtpd_recipient_restrictions=permit_mynetworks,reject
        -o smtpd_milters=
```
